src/fibonacci_n-step_number_sequences.rs

```diff
@@ -0,0 +1,68 @@
+// http://rosettacode.org/wiki/Fibonacci_n-step_number_sequences
+
+// stores current values, current sum, and position of least value (to be overwritten next)
+struct GenFibonacci {
+    buf:    Vec<u64>,
+    sum:    u64,
+    idx:    usize,
+}
+
+// iterator starts *after* the buffer contents
+impl Iterator for GenFibonacci {
+    type Item = u64;
+    fn next(&mut self) -> Option<u64> {
+        let result = Some(self.sum);
+        self.sum -= self.buf[self.idx];
+        self.buf[self.idx] += self.sum;
+        self.sum += self.buf[self.idx];
+        self.idx = (self.idx + 1) % self.buf.len();
+        result
+    }
+}
+
+// prints the starting buf and len number of additional elements
+fn print(buf: Vec<u64>, len: usize) {
+    let mut sum = 0;
+    for &elt in buf.iter() { sum += elt; print!("\t{}", elt); }
+    let iter = GenFibonacci { buf: buf, sum: sum, idx: 0 };
+    for x in iter.take(len) {
+        print!("\t{}", x);
+    }
+}
+
+fn test(mut buf: Vec<u64>, tgt: Vec<u64>) {
```

Comment:
You need a #[cgf(test)] here to avoid a warning (the function is only used when testing)